presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushLimitThroughProject.java

```diff
@@ -43,8 +42,6 @@
             return Optional.empty();
         }
 
-        return Optional.of(
-                replaceChildren(child, ImmutableList.of(
-                        replaceChildren(parent, child.getSources()))));
+        return Optional.of(transpose(parent, child));
```

Comment:
I would keep introduction of `transpose` method and refactor related to it as separate commit.